### PR TITLE
[IMP] xlsx: support import/export of data validation rules

### PR DESCRIPTION
**Pull request by Rachico**
This commit enhances XLSX handling capabilities by adding support for importing and exporting data validation rules, which helps to maintain data integrity and consistency when working with Excel files.

Task: [3637642](https://www.odoo.com/odoo/project/2328/tasks/3637642)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/xlsx/conversion/data_validation_conversion.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  };
+}
+
+function convertDateRule(id: number, dv: XLSXDataValidation): DataValidationRuleData {
+  const values = [dv.formula1.toString()];
+  if (dv.formula2 && dv.formula2 !== "undefined") {
+    values.push(dv.formula2.toString());
+  }
+  let criterion = {
+    type: XLSX_DV_DATE_OPERATOR_TO_DV_TYPE_MAPPING[dv.operator] ?? "dateIsBetween",
```
Comment (hokolomopo) [inline src/xlsx/conversion/data_validation_conversion.ts:100]: Should we really have a default ? The operator not being in the mapping mean that we do  not support it, we should skip the rule conversion altogether.

**Review by hokolomopo — COMMENTED** [review]
🏎️ 

I didn't test that it worked in excel, but I trust you taht it works. Somethign I feel like I already told you is that _**we should always avoid typecasting and ts-ignore**_.  Typescript is there for a reason, and he is usually correct, even if it's annoying. The only casting you need is the one in `convertDecimalCriterionTypeToExcelOperator` as we cannot type object keys, the other ones can be avoided I think.

Otherwise it looks fine 👋

File: tests/xlsx/xlsx_import.test.ts [hunk 2]
```diff
@@ -319,6 +320,86 @@ describe("Import xlsx data", () => {
...
+    const dvRule = getDataValidationBeginningAt(cellXc, testSheet);
+    switch (ruleType) {
+      case "number":
+      case "time":
+      case "textLength":
+        // Unsupported data validation type 'textLength'
+        expect(dvRule).toBeUndefined();
+        break;
+      case "decimal":
+        expect(dvRule).toMatchObject({
```
Comment (LucasLefevre) [inline tests/xlsx/xlsx_import.test.ts:344]: if you do a `.each` then switch/case on the value of the "each", then it should just be different tests. Even if it means repeating some setup lines

File: src/xlsx/functions/data_validation.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  const rule = dvRule.criterion;
+  const formula1 = adaptFormulaToExcel(rule.values[0]);
+  const formula2 = rule.values[1] ? adaptFormulaToExcel(rule.values[1]) : undefined;
+  const operator = convertDateCriterionTypeToExcelOperator(dvRule.criterion.type);
+  const attributes = commonDataValidationAttributes(dvRule);
+  attributes.push(["type", "date"], ["operator", operator]);
+  if (formula2) {
+    return escapeXml/*xml*/ `
+      <dataValidation ${formatAttributes(attributes)}>
+        <formula1>${toSerialDate(formula1)}</formula1>
```
[the remark this answers is not in the document]
Comment (Rachico) [reply src/xlsx/functions/data_validation.ts:63]: Excel expects a serial date to be able to preserve the correct date value.

**Review by LucasLefevre — APPROVED** [review]
robodoo r+

**Pull request closed without merging**